File: code/ch12/12.3.2.nimGame.py
class Solution:
    def canWinNim(self, n: int) -> bool:
        if n < 4:
            return True
        a, b, c = True, True, True
        for i in range(4, n + 1):
            current = not (a and b and c)
            a, b, c = b, c, current
        return c

File: code/ch12/12.3.2.nimGame.cpp
class Solution {
public:
    bool canWinNim(int n) {
        if (n < 4) return true;
        
        int a = true, b = true, c = true;
        
        for (int i = 4; i < n + 1; i++) {
            int current = !(a && b && c);
            a = b;
            b = c;
            c = current;
        }
        
        return c;
    }
};
